src/app/zap-templates/templates/app/im-cluster-command-handler.zapt

```diff
@@ -99,74 +99,79 @@ void Dispatch{{asUpperCamelCase side}}Command({{#if (isServer side)}}CommandHand
 
 } // namespace Clusters
 
-void DispatchSingleClusterCommand(ClusterId aClusterId, CommandId aCommandId, EndpointId aEndPointId,
-                             TLV::TLVReader & aReader, CommandHandler * apCommandObj)
+void SetInvalidClusterId(const ConcreteCommandPath & aCommandPath, Command * apCommandObj)
 {
-    ChipLogDetail(Zcl, "Received Cluster Command: Cluster=" ChipLogFormatMEI " Command=" ChipLogFormatMEI " Endpoint=%" PRIx16, ChipLogValueMEI(aClusterId),
-                  ChipLogValueMEI(aCommandId), aEndPointId);
-    Compatibility::SetupEmberAfObjects(apCommandObj, aClusterId, aCommandId, aEndPointId);
-    ConcreteCommandPath commandPath(aEndPointId, aClusterId, aCommandId);
+    ChipLogError(Zcl, "Unknown cluster " ChipLogFormatMEI, ChipLogValueMEI(aCommandPath.mClusterId));
+
+    // Unrecognized cluster ID, error status will apply.
+    CommandPathParams returnStatusParam = {
+        aCommandPath.mEndpointId,
+        0, // GroupId
+        aCommandPath.mClusterId,
+        aCommandPath.mCommandId,
+        (CommandPathFlags::kEndpointIdValid)
+    };
+
+    apCommandObj->AddStatusCode(
+      returnStatusParam,
+      Protocols::SecureChannel::GeneralStatusCode::kNotFound,
+      Protocols::SecureChannel::Id,
+      Protocols::InteractionModel::Status::InvalidCommand
```

Comment:
Per spec this should be an `UNSUPPORTED_CLUSTER` error.  And the protocol id should be IM, if the status is an IM status, right?

I guess you just moved this code, but please file a followup to fix it?

Reply:
As you guess, I just moved this code and reformat it a little bit as it was hard to read imho. I will open a followup of fix it directly, whatever is faster.

Reply:
I changed `Protocols::SecureChannel::Id` -> `Protocols::InteractionModel::Id` and `Protocols::InteractionModel::Status::InvalidCommand` -> `Protocols::InteractionModel::Status::UnsupportedCommand`